### PR TITLE
[Dataprotection] Onboard new RP Dataprotection to azure-cli-extensions

### PR DESCRIPTION
**Pull request by sambitratha**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Dataprotection

**Review by kairu-ms — COMMENTED** [review]
What is UpgradeLog.htm used for?

File: src/dataprotection/README.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    --resource-group "SampleResourceGroup" --vault-name "swaggerExample" 
+```
+##### Show #####
+```
+az dataprotection backup-vault show --resource-group "SampleResourceGroup" --vault-name "swaggerExample"
+```
+##### Show #####
+```
+az dataprotection backup-vault show --resource-group "SampleResourceGroup" --vault-name "swaggerExample"
+```
```
Comment (kairu-ms) [inline src/dataprotection/README.md:31]: They are duplicated.

File: src/dataprotection/README.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    --vault-name "swaggerExample" 
+```
+##### Show-in-resource-group #####
+```
+az dataprotection backup-vault show-in-resource-group --resource-group "SampleResourceGroup"
+```
+##### Show-in-subscription #####
+```
+az dataprotection backup-vault show-in-subscription
+```
```
Comment (kairu-ms) [inline src/dataprotection/README.md:44]: Please use 'show' command as a combination of `show-in-resource-group` and `show-in-subscription`

File: src/dataprotection/README.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    --operation-id "MjkxOTMyODMtYTE3My00YzJjLTg5NjctN2E4MDIxNDA3NjA2OzdjNGE2ZWRjLWJjMmItNDRkYi1hYzMzLWY1YzEwNzk5Y2EyOA==" \
+    --location "WestUS" 
+```
+#### dataprotection backup-vault-operation-result ####
+##### Show #####
+```
+az dataprotection backup-vault-operation-result show \
+    --operation-id "YWUzNDFkMzQtZmM5OS00MmUyLWEzNDMtZGJkMDIxZjlmZjgzOzdmYzBiMzhmLTc2NmItNDM5NS05OWQ1LTVmOGEzNzg4MWQzNA==" \
+    --resource-group "SampleResourceGroup" --vault-name "swaggerExample" 
+```
```
Comment (kairu-ms) [inline src/dataprotection/README.md:62]: The command `az dataprotection backup-vault-operation-result show` is strange. I think it should be `az dataprotection backup-vault show-operation-result`

File: src/dataprotection/README.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+#### dataprotection data-protection ####
+##### Check-feature-support #####
+```
+az dataprotection data-protection check-feature-support --location "WestUS" \
+    --parameters "{\\"featureType\\":\\"DataSourceType\\",\\"objectType\\":\\"FeatureValidationRequest\\"}" 
+```
```
Comment (kairu-ms) [inline src/dataprotection/README.md:68]: The command group `dataprotection data-protection` has duplications.

File: src/dataprotection/README.md [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+#### dataprotection export-job ####
+##### Trigger #####
+```
+az dataprotection export-job trigger --resource-group "SwaggerTestRg" --vault-name "NetSDKTestRsVault"
+```
+#### dataprotection export-job-operation-result ####
+##### Show #####
+```
+az dataprotection export-job-operation-result show --operation-id "00000000-0000-0000-0000-000000000000" \
+    --resource-group "SwaggerTestRg" --vault-name "NetSDKTestRsVault"
```
Comment (kairu-ms) [inline src/dataprotection/README.md:200]: `az dataprotection export-job show-operation-result` is better.

Comment (sambitratha) [reply src/dataprotection/README.md:200]: Hi @kairu-ms this file seems to be autogenerated and not hand written. And it's not exactly picking from the available list of commands. We have actually taken these feedback earlier and updated our readme.az.md to rename the cmdlets

**Review by kairu-ms — CHANGES_REQUESTED** [review]
There are some arguments such as --properties and --parameters are huge. I think you should flatten them. Please follow the [link](https://github.com/Azure/autorest.az#flatten-a-parameter) to config codegen.

File: src/dataprotection/README.md (hunk 5 above)
Comment (kairu-ms) [reply src/dataprotection/README.md:200]: Can you show me the  readme.az.md configuration. Let me check.

Comment (sambitratha) [reply src/dataprotection/README.md:200]: https://github.com/sambitratha/azure-rest-api-specs/blob/dpp-private/specification/dataprotection/resource-manager/readme.az.md

Comment (kairu-ms) [reply src/dataprotection/README.md:200]: Can you add the following configuration under `directive` to checkout whether rename works.

```
  - where:
      command: dataprotection export-job-operation-result show
    set:
      command: dataprotection export-job-operation-result show-operation-result
  - where:
      group: dataprotection export-job-operation-result
    set:
      group: dataprotection export-job
```

File: src/dataprotection/README.md (hunk 4 above)
Comment (sambitratha) [reply src/dataprotection/README.md]: updated the readme file

File: src/dataprotection/README.md (hunk 3 above)
Comment (sambitratha) [reply src/dataprotection/README.md]: updated readme file

File: src/dataprotection/README.md (hunk 2 above)
Comment (sambitratha) [reply src/dataprotection/README.md]: updated

File: src/dataprotection/README.md (hunk 1 above)
Comment (sambitratha) [reply src/dataprotection/README.md]: updated

**Comment by sambitratha** [comment]
@kairu-ms we have 2 command that expects parameters (since it is autogenerated). Our API request payload is quite complex and hence we decided to have separate initialize command to construct the request body and use the autogenerated command as it is to do the service request. Our initialize cmd will give a json output. User can save it in json file and give the filename as input to the commands which expects --parameters. Hope this clarifies

File: src/dataprotection/azext_dataprotection/manual/_params.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    get_job_status_values,
+    get_job_operation_values,
+    get_datasource_types
+)
+
+
+def load_arguments(self, _):
+    with self.argument_context('dataprotection backup-instance create') as c:
+        c.argument('backup_instance', type=validate_file_or_dict, help='Request body for operation Expected value: '
+                   'json-string/@json-file.')
```
Comment (kairu-ms) [inline src/dataprotection/azext_dataprotection/manual/_params.py:41]: This argument should be flatten as it's not polymorphic class. Below are the arguments generated by CodeGen, if you want to modify this command manually, you can copy the generated code.

```python

    with self.argument_context('dataprotection backup-instance create') as c:
        c.argument('vault_name', type=str, help='The name of the backup vault.')
        c.argument('resource_group_name', resource_group_name_type)
        c.argument('backup_instance_name', options_list=['--name', '-n', '--backup-instance-name'], type=str,
                   help='The name of the backup instance')
        c.argument('friendly_name', type=str, help='Gets or sets the Backup Instance friendly name.')
        c.argument('data_source_info', action=AddDataSourceInfo, nargs='+', help='Gets or sets the data source '
                   'information.')
        c.argument('data_source_set_info', action=AddDataSourceSetInfo, nargs='+', help='Gets or sets the data source '
                   'set information.')
        c.argument('object_type', type=str, help='')
        c.argument('policy_id', type=str, help='', arg_group='Policy Info')
        c.argument('policy_parameters', action=AddPolicyParameters, nargs='+', help='Policy parameters for the backup '
                   'instance', arg_group='Policy Info')
```

File: src/dataprotection/azext_dataprotection/manual/_params.py [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+    with self.argument_context('dataprotection backup-instance validate-for-backup') as c:
+        c.argument('backup_instance', type=validate_file_or_dict, help='Request body for operation Expected value: '
+                   'json-string/@json-file.')
```
Comment (kairu-ms) [inline src/dataprotection/azext_dataprotection/manual/_params.py:45]: Same concern as above.

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**